core/pom.xml

```diff
@@ -285,6 +285,9 @@
                                 <include>org/elasticsearch/cluster/routing/TestShardRouting.class</include>
                                 <include>org/elasticsearch/cluster/routing/TestShardRouting$*.class</include>
                                 <include>org/elasticsearch/index/shard/MockEngineFactoryPlugin.class</include>
+                                <!-- Shared between core and backwards compatibility tests. -->
+                                <include>org/elasticsearch/gateway/RecoveryFromGatewayIT.class</include>
```

Comment:
Maybe the shared portions should be refactored out? It seems odd to include an integration test in the test framework (I think it might be run from all integ tests then?)